deeplake/core/chunk_engine.py

```diff
@@ -714,63 +722,149 @@ def _samples_to_chunks(
         Returns:
             Tuple[List[BaseChunk], Dict[Any, Any]]
         """
+        extending = start_chunk_row is None and register
+        lengths = None
+        num_samples = self.num_samples
+        orig_meta_length = self.tensor_meta.length
+        incoming_num_samples = len(samples)
+        if extending:
+            num_samples = len(samples)
+            enc_ids = []
+            enc_count = [0]
```

Comment:
Instead of updating registering to the chunk id encoder after each chunk write, we store the updates in these lists and merge with existing encoder data in the end.